packages/fxa-auth-server/lib/db/index.js

```diff
@@ -548,28 +549,36 @@ module.exports = (config, log, Token, UnblockCode = null) => {
       return;
     }
 
-    let location;
-    if (geo && geo.location) {
-      location = {
-        city: geo.location.city,
-        country: geo.location.country,
-        countryCode: geo.location.countryCode,
-        state: geo.location.state,
-        stateCode: geo.location.stateCode,
+    let t;
+    if (onlyUpdateLastAccessTime) {
+      t = {
+        lastAccessTime: token.lastAccessTime,
+        id,
       };
-    }
+    } else {
+      let location;
+      if (geo && geo.location) {
+        location = {
+          city: geo.location.city,
+          country: geo.location.country,
+          countryCode: geo.location.countryCode,
+          state: geo.location.state,
+          stateCode: geo.location.stateCode,
+        };
+      }
 
-    const t = {
-      lastAccessTime: token.lastAccessTime,
-      location,
-      uaBrowser: token.uaBrowser,
-      uaBrowserVersion: token.uaBrowserVersion,
-      uaDeviceType: token.uaDeviceType,
-      uaFormFactor: token.uaFormFactor,
-      uaOS: token.uaOS,
-      uaOSVersion: token.uaOSVersion,
-      id,
-    };
+      t = {
+        lastAccessTime: token.lastAccessTime,
+        location,
+        uaBrowser: token.uaBrowser,
+        uaBrowserVersion: token.uaBrowserVersion,
+        uaDeviceType: token.uaDeviceType,
+        uaFormFactor: token.uaFormFactor,
+        uaOS: token.uaOS,
+        uaOSVersion: token.uaOSVersion,
+        id,
+      };
+    }
 
     return this.redis.touchSessionToken(uid, t);
```

Comment:
looking at `touchSessionToken_1.lua` it seems like having `t` exclude the location / ua fields will erase them from redis, is that true? If so we prob want to change the lua to merge the update.